src/supporting-files/stackbit-banner.html.js

```diff
@@ -53,52 +53,104 @@ module.exports = `<div id="theme-bar" class="theme-bar theme-bar-fixed theme-bar
-  var searchParams = window.location.search
-    .slice(1)
-    .split('&')
-    .reduce((res, cur) => {
-      const [key, value] = cur.split('=');
-      res[key] = value || true;
-      return res;
-    }, {});
-  var hideThemeBar = sessionStorage.getItem('hideThemeBar') || searchParams.themeBarHidden;
-  if (searchParams.themeBarTitle) {
-    themebar.querySelector('.theme-bar-name').textContent = decodeURIComponent(searchParams.themeBarTitle);
-  }
-  if (searchParams.themeBarFork) {
-    var forkButton = themebar.querySelector('.theme-bar-fork');
-    if (searchParams.themeBarFork === 'false') {
-      forkButton.style.display = 'none';
-    } else {
-      forkButton.querySelector('span').textContent = decodeURIComponent(searchParams.themeBarFork);
-    }
-  }
-  if (searchParams.themeBarNewSite) {
-    var newSiteButton = themebar.querySelector('.theme-bar-new-site');
-    if (searchParams.themeBarNewSite === 'false') {
-      newSiteButton.style.display = 'none';
-    } else {
-      newSiteButton.querySelector('span').textContent = decodeURIComponent(searchParams.themeBarNewSite);
-    }
-  }
-  if (searchParams.themeBarUserGroup) {
-    var newSiteButton = themebar.querySelector('.theme-bar-new-site');
-    var joiner = newSiteButton.search.length ? '&' : '';
-    newSiteButton.search += joiner + 'userGroup=' + searchParams.themeBarUserGroup;
-  }
+  const body = document.querySelector("body");
+  const themebar = document.querySelector("#theme-bar");
+  const searchParams = new URLSearchParams(window.location.search);
+
+  // Hide themebar from localstorage
+  const hideThemeBar = sessionStorage.getItem("hideThemeBar") || searchParams.get("themeBarHidden");
   if (body && !hideThemeBar) {
     body.classList.add("has-theme-bar");
     themebar.classList.remove("theme-bar-hidden");
   }
-  document
-    .querySelector("#remove-theme-bar")
-    .addEventListener("click", function(e) {
-      e.preventDefault();
-      body.classList.remove("has-theme-bar");
-      themebar.classList.add("theme-bar-hidden");
-      sessionStorage.setItem('hideThemeBar', true);
-    });
+
+  const demo = searchParams.get("demo")
+  let themeBarProperties = {}
+
+  // themeBarConfig is loaded from themeBarConfig.js in the <head> of this document. The file is generated seperately in \`stackbit-themes\` repo at build time.
+  if (demo && themeBarConfig) {
+    if (themeBarConfig[demo]) {
+      themeBarProperties = themeBarConfig[demo]
+    }
+  } else {
+    themeBarProperties = {
+      themeBarTitle: searchParams.get("themeBarTitle"),
+      themeBarFork: searchParams.get("themeBarFork"),
+      themeBarForkLink: searchParams.get("themeBarForkLink"),
+      themeBarNewSite: searchParams.get("themeBarNewSite"),
+      themeBarNewSiteLink: searchParams.get("themeBarNewSiteLink"),
+      themeBarNewSiteTheme: searchParams.get("themeBarNewSiteTheme"),
+      themeBarUserGroup: searchParams.get("themeBarUserGroup")
```

Comment:
@JugglerX @VitaliyR please rename this field to something else.
For my understanding, it is needed for an extra parameter like `userGroup=nocode`.
If this is the case please provide different parameter like: `url_params=${encodeUrlComponent('userGroup=nocode')}`

The idea is not to have parameter names specific to stackbit internal features as this is an open-source project.

Reply:
removed. this is now handled by `themeBarNewSiteLinkParams` or can be set in the config (which is even better)